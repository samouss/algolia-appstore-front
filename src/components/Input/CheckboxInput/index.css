.CheckboxInput {
  position: relative;
  overflow: hidden;
}

.CheckboxInput__Input {
  position: absolute;
  left: -99999px;
}

.CheckboxInput__Label {
  width: 100%;
  padding: 5px 10px;

  &:hover {
    cursor: pointer;
    color: #0080ff;
  }
}

.CheckboxInput__Label--active {
  color: #0080ff;

  &:hover {
    text-decoration: line-through;
  }
}
