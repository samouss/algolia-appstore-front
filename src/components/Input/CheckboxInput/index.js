import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './index.css';

const CheckboxInput = ({
  id,
  value,
  checked,
  onChange,
  children,
  customClassName,
  customActiveClassName,
}) => (
  <div styleName="CheckboxInput">
    <input
      styleName="CheckboxInput__Input"
      id={id}
      value={value}
      type="checkbox"
      onChange={onChange}
      checked={checked}
    />
    <label
      htmlFor={id}
      className={cx(customClassName, {
        [customActiveClassName]: checked,
      })}
      styleName={cx('CheckboxInput__Label', {
        'CheckboxInput__Label--active': checked,
      })}
    >
      {children}
    </label>
  </div>
);

CheckboxInput.propTypes = {
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  customClassName: PropTypes.string,
  customActiveClassName: PropTypes.string,
};

CheckboxInput.defaultProps = {
  customClassName: '',
  customActiveClassName: '',
};

export default CheckboxInput;
