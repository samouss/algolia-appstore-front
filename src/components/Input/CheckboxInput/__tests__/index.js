import React from 'react';
import { shallow } from 'enzyme';
import CheckboxInput from '../index';

describe('<CheckboxInput />', () => {
  const defaultProps = {
    id: 'category-games',
    name: 'category',
    value: 'games',
    checked: false,
    onChange: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <CheckboxInput
        {...props}
      >
        label
      </CheckboxInput>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render when checked', () => {
    const props = {
      ...defaultProps,
      checked: true,
    };

    const component = shallow(
      <CheckboxInput
        {...props}
      >
        label
      </CheckboxInput>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render with customClassName', () => {
    const props = {
      ...defaultProps,
      customClassName: 'custom-class-name',
    };

    const component = shallow(
      <CheckboxInput
        {...props}
      >
        label
      </CheckboxInput>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onChange on input change', () => {
    const props = {
      ...defaultProps,
      onChange: jest.fn(),
    };

    const component = shallow(
      <CheckboxInput
        {...props}
      >
        label
      </CheckboxInput>,
    );

    component
      .find('input')
      .simulate('change');

    expect(props.onChange).toHaveBeenCalled();
  });
});
