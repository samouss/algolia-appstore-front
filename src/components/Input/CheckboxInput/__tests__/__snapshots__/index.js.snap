// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxInput /> expect to render 1`] = `
<div
  styleName="CheckboxInput"
>
  <input
    checked={false}
    id="category-games"
    onChange={[Function]}
    styleName="CheckboxInput__Input"
    type="checkbox"
    value="games"
  />
  <label
    className=""
    htmlFor="category-games"
    styleName="CheckboxInput__Label"
  >
    label
  </label>
</div>
`;

exports[`<CheckboxInput /> expect to render when checked 1`] = `
<div
  styleName="CheckboxInput"
>
  <input
    checked={true}
    id="category-games"
    onChange={[Function]}
    styleName="CheckboxInput__Input"
    type="checkbox"
    value="games"
  />
  <label
    className=""
    htmlFor="category-games"
    styleName="CheckboxInput__Label CheckboxInput__Label--active"
  >
    label
  </label>
</div>
`;

exports[`<CheckboxInput /> expect to render with customActiveClassName 1`] = `
<div
  styleName="CheckboxInput"
>
  <input
    checked={true}
    id="category-games"
    onChange={[Function]}
    styleName="CheckboxInput__Input"
    type="checkbox"
    value="games"
  />
  <label
    className="custom-active-class-name"
    htmlFor="category-games"
    styleName="CheckboxInput__Label CheckboxInput__Label--active"
  >
    label
  </label>
</div>
`;

exports[`<CheckboxInput /> expect to render with customClassName 1`] = `
<div
  styleName="CheckboxInput"
>
  <input
    checked={false}
    id="category-games"
    onChange={[Function]}
    styleName="CheckboxInput__Input"
    type="checkbox"
    value="games"
  />
  <label
    className="custom-class-name"
    htmlFor="category-games"
    styleName="CheckboxInput__Label"
  >
    label
  </label>
</div>
`;
